src/h5web/visualizations/heatmap/DomainSlider.tsx

```diff
@@ -35,16 +36,16 @@ function DomainSlider(): JSX.Element {
       renderThumb={(props, state) => (
         <div {...props}>{format(`.${MIN_DECIMALS}f`)(state.valueNow)}</div>
       )}
-      defaultValue={customDomain}
-      min={extendedDomain[0]}
-      max={extendedDomain[1]}
+      value={customDomain}
+      min={extendedMin}
+      max={extendedMax}
       step={step}
       pearling
       onAfterChange={values => {
-        const roundedValues = (values as Domain).map(value =>
+        const roundedDomain = (values as number[]).map(value =>
           round(value, MIN_DECIMALS)
-        );
-        setCustomDomain(roundedValues as Domain);
+        ) as Domain;
+        setCustomDomain(roundedDomain);
```

Comment:
Thanks for the clean-up :wink:

Reply:
No worries. The important change is `value` instead of `defaultValue`. With `defaultValue`, the change of custom domain when changing dataset was not being processed by the slider.